source/extensions/filters/http/istio_stats/istio_stats.cc

```diff
@@ -761,6 +762,34 @@ class IstioStatsFilter : public Http::PassThroughFilter,
         service_host_name = service_host;
       }
     }
+
+    // Implements fallback from using the namespace from SAN if available to
```

Comment:
In the existing implementation without this PR, where does SAN come from (e.g., from peer metadata or from peer certificate)? Is peer metadata from metadata exchange or from the metadata in endpoint of an EDS response?

Reply:
SAN always comes from a certificate. Peer metadata is from metadata exchange. EDS response only contains destination service I think.